### PR TITLE
Duckdns alias support

### PR DESCRIPTION
**Pull request by noamokman**
Hey, rebased #521 let me know what else needs to be done to support this

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @noamokman, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;hassio-addons#1253).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by noamokman** [comment]
@pvizeli would love your input

**Comment by noamokman** [comment]
Added the new option to the readme

File: duckdns/README.md [hunk 1]
```diff
@@ -89,6 +89,17 @@ The DuckDNS authentication token found at the top of the DuckDNS account landing
...
+### Option: `aliases` (optional)
+
+A list aliases of domains configured on the `domains` option.
+This is useful in cases where you would like to use your own domain.
+Create a CNAME record to point at the DuckDNS subdomain and set this value accrodingly.
+
+For example:
+```json
+[{"domain": "my-domain.duckdns.org", "alias": "ha.my-domain.com"}]
+```
```
Comment (pvizeli) [inline duckdns/README.md:101]: YAML please

Comment (noamokman) [reply duckdns/README.md]: Fixed 👍

**Comment by noamokman** [comment]
I'll try and test it on my server

**Comment by noamokman** [comment]
OK, tested it here:
https://github.com/noamokman/hassio-repository
It works, but the configuration to make it work is a bit confusing.
In my system it looks like this:
```yaml
domains:
  - mydns.duckdns.org
  - my-custom-domain.com
aliases:
  - domain: my-custom-domain.com
    alias: mydns.duckdns.org
```
This way the certificate holds both domains and the dns challenge succeeds

**Comment by noamokman** [comment]
Another cool tip, I use the nginx proxy, all I had to do is add the new name to the domain in the configuration:
```yaml
# before
domain: mydns.duckdns.org
# after
domain: mydns.duckdns.org my-custom-domain.com
```
That way, both urls still work, with ssl
Should I add it in the docs here?

**Comment by noamokman** [comment]
Thanks!

**Comment by TheGroundZero** [comment]
Does this properly request SSL certs for the alias domains, when they need a specific config?

E.g. a domain hosted at OVH uses the dns-ovh provider with the DNS challenge in the LetsEncrypt plugin.
This provider needs API credentials to be able to create a verification subdomain.  
There's no option (afaik) to add these credentials or to define that a certain domain/alias requires this verification method inside the DuckDNS plugin.

**Comment by noamokman** [comment]
I Don't think so, I just forwarded the subdomain for the challenge

**Comment by Pabsilon** [comment]
> OK, tested it here:
> https://github.com/noamokman/hassio-repository
> It works, but the configuration to make it work is a bit confusing.
> In my system it looks like this:
> 
> ```yaml
> domains:
>   - mydns.duckdns.org
>   - my-custom-domain.com
> aliases:
>   - domain: my-custom-domain.com
>     alias: mydns.duckdns.org
> ```
> 
> This way the certificate holds both domains and the dns challenge succeeds

I'm doing this same configuration and the dns challenge succeeds, but the duckDNS addon seems to have an error when updating the duckdns.org domain IP.
When I remove my custom domain from the domains list, it works correctly, so it looks like the addon is trying to update my custom domain on duckdns.org.

**Pull request merged**